lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -109,6 +109,11 @@ def add_metadata(post, options)
 
       def add_payment_method(post, payment_method, options)
         post[:source] = {}
+        if options[:token_type] == 'id'
```

Comment:
can you explain this `id` concept?

Reply:
I feel it's a concept very specific to checkout maybe? let's generalize it.

Reply:
maybe we can create something called PSPTokenizedCard and use type as `psp_token` to determine this?

Reply:
I created a model `TokenizedCard` and kept the payment_data in it as a dict for now. And used the model for the check here